src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -505,29 +515,41 @@ private static bool LineIsEmpty(ArraySegment<byte> line)
 
         private void ParseStatusLine(Span<byte> line, HttpResponseMessage response)
         {
+            // We sent the request version as either 1.0 or 1.1.
+            // We expect a response version of the form 1.X, where X is a single digit as per RFC.
+
             if (line.Length < 14 || // "HTTP/1.1 123\r\n" with optional phrase before the crlf
                 line[0] != 'H' ||
                 line[1] != 'T' ||
                 line[2] != 'T' ||
                 line[3] != 'P' ||
                 line[4] != '/' ||
+                line[5] != '1' ||
+                line[6] != '.' ||
                 line[8] != ' ')
```

Comment:
After this change, we should look at whether it's more efficient to simply reinterpret the first 8 bytes of the line as a long and then just do a comparison against an equivalent for the bytes for HTTP/1.0 and HTTP/1.1, e.g.
```C#
private static readonly ulong s_http10Bytes = BitConverter.ToUInt64(Encoding.ASCII.GetBytes("HTTP/1.0"), 0);
private static readonly ulong s_http11Bytes = BitConverter.ToUInt64(Encoding.ASCII.GetBytes("HTTP/1.1"), 0);
...
if (line.Length < 14 || // "HTTP/1.1 123\r\n" with optional phrase before the crlf
    line[8] != ' ')
{
    ThrowInvalidHttpResponse();
}

ulong first8Bytes = BitConverter.ToInt64(line.Slice(0, sizeof(long));
if (first8Bytes != s_http10Bytes && first8Bytes != s_http11Bytes)
{
   ThrowInvalidHttpResponse();
}
```

Reply:
I'm reluctant to make to make this change without analyzing the perf, which I don't want to spend time on right now.  Good issue to investigate for the future though.

Reply:
Yup.